tasks/brolly_stats_rollup.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import datetime
+import json
+import logging
+import uuid  # TODO remove
+
+import httpx
+from shared.celery_config import brolly_stats_rollup_task_name
+from shared.config import get_config
+from shared.metrics import metrics
+
+from app import celery_app
+from database.models import Commit, Constants, Repository, Upload, User
+from tasks.crontasks import CodecovCronTask
+
+log = logging.getLogger(__name__)
+
+DEFAULT_BROLLY_ENDPOINT = "https://codecov.io/self-hosted/telemetry"
+
+
+class BrollyStatsRollupTask(CodecovCronTask):
+    """
+    By default, this cron task collects anonymous statistics about the Codecov instance
+    and submits them to Codecov to give us insight into the size of our self-hosted,
+    open-source userbase.
+
+    Installations can configure the behavior in `codecov.yml`:
```

Comment:
Oh this is nice. Having explanation of the config associated with some part of the code directly in the code. 10/10, thanks for doing this.